pywbem/_units.py

```diff
@@ -0,0 +1,749 @@
+version 2.8, and the following additional ``Units`` qualifier values that are
+used in DMTF CIM Schema version 2.49:
+
++--------------------------------------+
+| Additional ``Units`` values          |
++======================================+
+| ``-dBm``                             |
++--------------------------------------+
+| ``Blocks``                           |
++--------------------------------------+
+| ``Percentage``                       |
++--------------------------------------+
+| ``Proportion``                       |
++--------------------------------------+
+| ``Tenths of Revolutions per Minute`` |
++--------------------------------------+
+
+By default, the string value returned from these functions may contain the
+following Unicode characters outside of the 7-bit ASCII range. If the
+``use_ascii`` parameter is `True`, these Unicode characters are replaced with
+7-bit ASCII text as follows:
+
++---------------------------+---------+---------------+
+| Unicode character code    | Unicode | 7-bit ASCII   |
++===========================+=========+===============+
+| U+00B0: DEGREE SIGN       | ``°``   | ``deg``       |
++---------------------------+---------+---------------+
+| U+00B5: MICRO SIGN        | ``µ``   | ``u``         |
++---------------------------+---------+---------------+
+| U+2030: PER MILLE SIGN    | ``‰``   | ``1/1000``    |
++---------------------------+---------+---------------+
+| U+2126: OHM SIGN          | ``Ω``   | ``Ohm``       |
++---------------------------+---------+---------------+
+| U+00B2: SUPERSCRIPT TWO   | ``²``   | ``^2``        |
++---------------------------+---------+---------------+
+| U+00B3: SUPERSCRIPT THREE | ``³``   | ``^3``        |
++---------------------------+---------+---------------+
+
+Examples:
+
+* ``PUnit("byte / second * 10^3")``-> ``kB/s``
+* ``PUnit("byte * 2^10")``-> ``KiB``
+* ``PUnit("hertz * 10^6")``-> ``MHz``
+* ``PUnit("ampere * 10^-3")`` -> ``mA``
+* ``Units("KiloBits per Second")`` -> ``kbit/s``
+* ``Units("Tenths of Degrees C")`` -> ``1/10 °C``
+
+Limitations:
+
+* For PUnit qualifiers, vendor-defined base units are not supported
+  (e.g. ``vendor:myunit``).
+
+* For PUnit qualifiers, space characters within the parenthesis of
+  ``decibel`` (e.g. ``decibel ( A )``) are not supported.
+
+* For Units qualifiers, arbitrary numeric values that are part of the Units
+  value (e.g. ``<numeric-value> NanoSeconds`` or
+  ``Amps at <numeric-value> Volts``) are not generally supported, but
+  only for those cases that are used in the DMTF CIM Schema (as of its version
+  2.49):
+
+  - ``250 NanoSeconds``
+  - ``Amps at 120 Volts``
+"""
+
+import re
+try:
+    from collections.abc import Sequence
+except ImportError:
+    from collections import Sequence
+from nocasedict import NocaseDict
+import six
+
+from ._cim_obj import CIMProperty, CIMMethod, CIMParameter
+from ._utils import _ensure_unicode
+
+__all__ = ['siunit_obj', 'siunit']
```

Comment:
Should we put the public functions at the top of the module and the supporting ones further down in the code since it is primarily the public modules that people will look at.

Reply:
Moved them to the top.